src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
There is nothing that ensures this at the moment. In the case that you create a `Realm` and its `SharedGroup` with a `ChangesetCookeer` then reset the `shared_ptr` or SyncConfig` after creation then you would run into issues. This would pretty much be going out of your way to break things though so not sure this is a case we need to deal with - if we do need to prevent this we can either store the `ChangesetCooker` separately from `Config` in the `Realm` object, or we should try to make it const (not sure if this is possible). 

The only other possible issue I can think of would be during destruction as both the `ChangesetCooker` and `SharedGroup` are destroyed in the `Realm` destructor - if the cooker is destroyed first and things are happening on other threads this might cause issues but not sure if this is something we need to worry about or deal with.